### PR TITLE
Add $database parameter for methods: listTableIndexes, listTableDetails

### PR DESCRIPTION
**Pull request by stdex**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no  
| Fixed issues | #3611  

#### Summary

Rebase pull request: https://github.com/doctrine/dbal/pull/885

### CODE REVIEW

**Review by SenseException — CHANGES_REQUESTED** [review]
Please also add `?string` type to the added `$database` arguments instead of just the phpdocs.

If your commits aren't separated logical steps, please squash your commits into one or either give them a better message instead of using the same commit message for all 4 commits.

File: lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php [hunk 1]
```diff
@@ -174,13 +174,18 @@ public function listTableColumns($table, $database = null)
...
      * Keys of the portable indexes list are all lower-cased.
      *
-     * @param string $table The name of the table.
+     * @param string      $table    The name of the table.
+     * @param string|null $database The name of the database
      *
      * @return Index[]
      */
-    public function listTableIndexes($table)
+    public function listTableIndexes(string $table, ?string $database = null)
```
Comment (greg0ire) [inline lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php:182]: This is a BC-break for extending classes, I think you will have to resort to `func_get_args` if you want to make this BC.

Comment (SenseException) [reply lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php:182]: True, the second argument is a BC break. :disappointed:

**Review by morozov — COMMENTED** [review]
Sorry, I thought I posted this 3 days ago but I didn't.

**Pull request closed without merging**